jems-testing/src/main/java/org/dmfs/jems/pair/hamcrest/PairMatcher.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.pair.hamcrest;
+
+import org.dmfs.jems.pair.Pair;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+
+/**
+ * {@link Matcher} for {@link Pair}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PairMatcher<L, R> extends TypeSafeDiagnosingMatcher<Pair>
+{
+    private final Matcher<L> mLeftValueMatcher;
+    private final Matcher<R> mRightValueMatcher;
+
+
+    public PairMatcher(Matcher<L> leftValueMatcher, Matcher<R> rightValueMatcher)
+    {
+        mLeftValueMatcher = leftValueMatcher;
+        mRightValueMatcher = rightValueMatcher;
+    }
+
+
+    @Override
+    protected boolean matchesSafely(Pair actualPair, Description mismatchDescription)
+    {
+        if (!mLeftValueMatcher.matches(actualPair.left()))
+        {
+            mismatchDescription.appendText("Left value doesn't match: ");
+            mLeftValueMatcher.describeMismatch(actualPair.left(), mismatchDescription);
+            return false;
+        }
+        else if (!mRightValueMatcher.matches(actualPair.right()))
+        {
+            mismatchDescription.appendText("Right value doesn't match: ");
+            mLeftValueMatcher.describeMismatch(actualPair.right(), mismatchDescription);
```

Comment:
still using `mLeftValueMatcher`, should be `mRightValueMatcher`.

Reply:
oh, I only corrected the the argument..  fixed now